proto-public/pbmesh/v1alpha1/upstreams_configuration.proto

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v1alpha1;
+
+import "google/protobuf/duration.proto";
+import "pbcatalog/v1alpha1/protocol.proto";
+import "pbcatalog/v1alpha1/selector.proto";
+import "pbmesh/v1alpha1/connection.proto";
+import "pbmesh/v1alpha1/routing.proto";
+import "pbresource/resource.proto";
+
+message UpstreamsConfiguration {
+  // Selection of workloads these upstreams should apply to.
+  // These can be prefixes or specific workload names.
+  hashicorp.consul.catalog.v1alpha1.WorkloadSelector workloads = 1;
+
+  // default_config applies to all upstreams for the workloads selected by this resource.
+  UpstreamConfig default_config = 2;
+
+  // config_overrides provides per-upstream or per-upstream-port config overrides.
+  repeated UpstreamConfigOverrides config_overrides = 3;
+}
+
+message UpstreamConfigOverrides {
```

Comment:
I wonder if it's worth being more explicit here about how the first three fields are "key-like" in that they are used like a map key to determine which `UpstreamConfig` in the overrides slice is the one to use.

Reply:
Added a comment to this message